components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -152,64 +152,44 @@ public Response issueAccessToken(@Context HttpServletRequest request,
                     return handleBasicAuthFailure();
                 }
             }
+            CarbonOAuthTokenRequest oauthRequest = null;
 
             try {
-                CarbonOAuthTokenRequest oauthRequest = new CarbonOAuthTokenRequest(httpRequest);
-                // exchange the access token for the authorization grant.
-                OAuth2AccessTokenRespDTO oauth2AccessTokenResp = getAccessToken(oauthRequest);
-                // if there BE has returned an error
-                if (oauth2AccessTokenResp.getErrorMsg() != null) {
-                    // if there is an auth failure, HTTP 401 Status Code should be sent back to the client.
-                    if (OAuth2ErrorCodes.INVALID_CLIENT.equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleBasicAuthFailure();
-                    } else if ("sql_error".equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleSQLError();
-                    } else if (OAuth2ErrorCodes.SERVER_ERROR.equals(oauth2AccessTokenResp.getErrorCode())) {
-                        return handleServerError();
-                    } else {
-                        // Otherwise send back HTTP 400 Status Code
-                        OAuthResponse.OAuthErrorResponseBuilder oAuthErrorResponseBuilder = OAuthASResponse
-                                .errorResponse(HttpServletResponse.SC_BAD_REQUEST)
-                                .setError(oauth2AccessTokenResp.getErrorCode())
-                                .setErrorDescription(oauth2AccessTokenResp.getErrorMsg());
-                        OAuthResponse response = oAuthErrorResponseBuilder.buildJSONMessage();
-
-                        ResponseHeader[] headers = oauth2AccessTokenResp.getResponseHeaders();
-                        ResponseBuilder respBuilder = Response
-                                .status(response.getResponseStatus());
-
-                        if (headers != null && headers.length > 0) {
-                            for (int i = 0; i < headers.length; i++) {
-                                if (headers[i] != null) {
-                                    respBuilder.header(headers[i].getKey(), headers[i].getValue());
-                                }
-                            }
-                        }
+                oauthRequest = new CarbonOAuthTokenRequest(httpRequest);
+            } catch (OAuthProblemException e) {
+                if (log.isDebugEnabled()) {
```

Comment:
Few things .....

Invalid input is not exceptional. So they shouldn't be handled as exceptions.
But if a third party library handles it that way, we can't help and we should handle the exception.

But if it's a generic exception then there is a problem. Because the exception can be thrown for other cases as well.

In those cases, we should conditionally suppress the stack traces only for invalid user inputs.